EIPS/eip-6051.md

```diff
@@ -0,0 +1,469 @@
+```
+// s same as Case 1 and Case 2
+s = '0x28fa2db9f916e44fcc88370bedaf5eb3ec45632f040f4c1450c0f101e1e8bac8',
+S = '0xd2fd6fcaac231d08363e736e61edb7e7696b13a727e3d2a239415cb8dc6ee278'
+```
+The shared secret, symmetric key and IV should be:
+```
+ss: '0xe0b36f56cdb63c27e933a5a67a5e97db4b566c9276a36aeee5dc6e87da118867',
+skey: '0x7c6fa749e6df13c8578dc44cb24cdf46a44cb163e1e570c2e590c720aed5783f',
+IV: '0x3c98ef6fc34b0d6e7e16bd78'
+```
+Then the return value should be `S || cipher`:
+```
+'0xd2fd6fcaac231d08363e736e61edb7e7696b13a727e3d2a239415cb8dc6ee2786a7e2e40efb86dc68f44f3e032bbedb1259fa820e548ac5adbf191784c568d4f642ca5b60c0b2142189dff6ee464b95c'
+```
+
+Then **R3** is provided as:
+```javascript
+request({
+	method: 'eth_intakePrivateKey',
+	params: [
+		version: 'Curve25519-Chacha20-Poly1305',
+		recipientPublicKey: '0xc0ea3514b0ab83b2fe4f4ef96159cda8fa836ce549ef09569b901eef0723bf79',
+		oob: '0x313233343536',
+		salt: '0x6569703a2070726976617465206b657920656e63617073756c6174696f6e',
+		data: '0xd2fd6fcaac231d08363e736e61edb7e7696b13a727e3d2a239415cb8dc6ee2786a7e2e40efb86dc68f44f3e032bbedb1259fa820e548ac5adbf191784c568d4f642ca5b60c0b2142189dff6ee464b95c'
+	],
+})
+```
+The return value should be `0x001d3f1ef827552ae1114027bd3ecf1f086ba0f9`. This matches the `account` parameter in **R2**.
+
+## Reference Implementation
+
+Contributions are welcome.
+
+Test vector generator is provided earlier in the **Test Cases** section.
+
+## Security Considerations
+
+**Perfect Forward Secrecy**: PFS is achieved by using ephemeral key pairs in both sides.
+
+**Optional Signature and Trusted Public Keys**
+
+`R` could be signed so that Sender application can verify if `R` could be trusted or not. This involves both signature verification and if the signer could be trusted or not. While signature verification is quite straightforward in itself, the latter should be managed with care. To facilitate this trust management issue, `signerPubKey` could be further signed, creating a dual-layer trust structure:
+
+```
+R <-- signerPubKey <-- trustedPubKey
+```
+
+This allows various strategies to mange the trust. For example:
+
+-   A hardware wallet vendor which takes it very serious about the brand reputation and the fund safety for its customers, could choose to trust only its own public keys, all instances of `trustedPubKey`. These public keys only sign `signerPubKey` from selected partners.
+-   A MPC service could publish its `signerPubKey` online so that Sender application won't verify the signature against a wrong or fake public key.
+
+Note that it is advised that a separate key pair should be used for signing on each curve.
+
+**Security Level**:
+
+1. We are not considering post-quantum security. If quantum computer becomes a materialized threat, the underlying cipher of Ethereum and other L1 chains would have been replaced, and this EIP will be outdated then (as the EC part of ECIES is also broken).
```

Comment:
```suggestion
1. We are not considering post-quantum security. If the quantum computer becomes a materialized threat, the underlying cipher of Ethereum and other L1 chains would have been replaced, and this EIP will be outdated then (as the EC part of ECIES is also broken).
```

Reply:
I think this is generally omittable because the whole Ethereum rely on ECDSA / secp256k1, if ECDSA is no longer safe due to quantum security, there are much bigger threat than the encapsulation described here.

Reply:
I'd like to retain this part as part of the justifications for the current design.